src/app/modules/gb-analysis-module/fractalis-components/gb-fractalis-chart/gb-fractalis-chart.component.ts

```diff
@@ -1,37 +1,128 @@
-import {Component, Input, OnInit} from '@angular/core';
+/**
+ * Copyright 2017 - 2018  The Hyve B.V.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {AfterViewInit, Component, Input} from '@angular/core';
 import {Chart} from '../../../../models/chart-models/chart';
 import {ChartType} from '../../../../models/chart-models/chart-type';
 import {FractalisService} from '../../../../services/fractalis.service';
+import {FractalisChartDescription} from '../../../../models/fractalis-models/fractalis-chart-description';
+import {FractalisData} from '../../../../models/fractalis-models/fractalis-data';
 import {Concept} from '../../../../models/constraint-models/concept';
+import {ChartValidator} from '../../../../utilities/chart-validator';
 
 @Component({
   selector: 'gb-fractalis-chart',
   templateUrl: './gb-fractalis-chart.component.html',
   styleUrls: ['./gb-fractalis-chart.component.css']
 })
-export class GbFractalisChartComponent implements OnInit {
+export class GbFractalisChartComponent implements AfterViewInit {
 
   @Input() chart: Chart;
+  private fractalisChartDescription: FractalisChartDescription;
+  private fractalisChart: object;
+  private chartValidator: ChartValidator;
 
   constructor(private fractalisService: FractalisService) {
+    this.chartValidator = new ChartValidator();
   }
 
-  ngOnInit() {
+  ngAfterViewInit() {
     this.initializeChart();
   }
 
-  initializeChart() {
-    // TODO 1: modify fractalis.js so that it is responsive to gridster layout
-    // TODO 2: figure out an async detector
-    // that detects when the chart div is available to fractalis, ngOnInit is still too early
-    this.fractalisService.F.setChart('histogram', '#' + this.chart.id);
-    // fractal.getChartParameterDescriptions(this.myChart1, (v) => {
-    //   console.log('v', v)
-    //   // fractal.setChartParameters(myChart1, v);
-    // });
+  private initializeChart() {
+    if (this.chart.type !== ChartType.CROSSTABLE) {
+      this.fractalisChart = this.fractalisService.setChart('#' + this.chart.id);
+      this.fractalisService.F.getChartParameterDescriptions(this.fractalisChart, (description: FractalisChartDescription) => {
+          this.fractalisChartDescription = description;
+          this.chartValidator.updateValidationCriteria(description.catVars, description.numVars);
+          console.log(`Chart ${this.chart.id}`, this.fractalisChartDescription);
+          this.setVariablesIfValid();
+        },
+        (err) => console.log(`Failed to add a chart with id ${this.chart.id}. ${err}`)
+      );
+    }
+    console.log(`Added a new chart with id ${this.chart.id}`);
+  }
+
+  setVariablesIfValid() {
```

Comment:
consider separating validating variables out, e.g. validateVariables()

Reply:
hmm, but setVariablesIfValid is basically `if valid => setVariables, if not => invalidate`
`validateVariables` is named `isNumberOfVariablesValid`
You mean different naming or different separation?